### PR TITLE
Refactoring of client side code

### PR DESCRIPTION
**Pull request by clochix**
There should be no visible effect.

### CODE REVIEW

**Comment by clochix** [comment]
Ping @aenario

File: client/app/components/account-config.coffee [hunk 1]
```diff
@@ -688,8 +688,13 @@ AccountConfigMailboxes = React.createClass
 
     render: ->
         favorites = @state.favoriteMailboxes.value
+        mailboxesFlat = {}
         if @state.mailboxes.value isnt '' and favorites isnt ''
             mailboxes = @state.mailboxes.value.map (mailbox, key) =>
+                id = mailbox.get 'id'
+                mailboxesFlat[id] = {}
```
Comment (aenario) [inline client/app/components/account-config.coffee]: move this to an helper function

File: client/app/components/account-config.coffee [hunk 2]
```diff
@@ -771,6 +776,13 @@ AccountConfigMailboxes = React.createClass
 
     _renderMailboxChoice: (labelText, box) ->
         if @state.id? and @state.mailboxes.value isnt ''
+            mailboxesFlat = {}
+            @state.mailboxes.value.map (mailbox) ->
+                id = mailbox.get 'id'
+                mailboxesFlat[id] = {}
```
Comment (aenario) [inline client/app/components/account-config.coffee]: move this to an helper function

Comment (aenario) [inline client/app/components/account-config.coffee]: Or may be to the state itself ? This way this part can stay unmutated. 
Why is selectedID copied to each mailbox ?

File: client/app/components/application.coffee [hunk 3]
```diff
@@ -406,13 +414,33 @@ module.exports = Application = React.createClass
         else
             selectedMailboxID = null
 
+        accounts = AccountStore.getAll()
+        accountsFlat = {}
```
Comment (aenario) [inline client/app/components/application.coffee]: idem

File: client/app/components/mails_input.coffee [hunk 4]
```diff
@@ -159,7 +159,7 @@ module.exports = MailsInput = React.createClass
                 "#{contact.get 'fn'} <#{contact.get 'address'}>"
 
     onQuery: (char) ->
-        query = @refs.contactInput.getDOMNode().value.split(',').pop().trim()
+        query = @refs.contactInput.getDOMNode().value.split(',').pop().trimLeft()
```
Comment (aenario) [inline client/app/components/mails_input.coffee]: is this standard ?

File: server/models/requests.coffee [hunk 5]
```diff
@@ -10,7 +10,7 @@ module.exports =
 
     contact:
         all: cozydb.defaultRequests.all
-        byName: (doc) ->
+        mailByName: (doc) ->
```
Comment (aenario) [inline server/models/requests.coffee]: DS should handles conflict resolution. Not sure if we should also prefix in each app for additional safety in dev mode. @poupotte @frankrousseau , any opinion ?

**Comment by aenario** [comment]
Please add some comments regarding the flattening + de-immutabilisation of parameters.  

I am not sure if your request name changes on server-side are necessary / a good idea. May be we just want to enable the DS request conflict resolution in NODE_ENV=development...

File: client/app/components/mails_input.coffee (hunk 4 above)
Comment (clochix) [inline client/app/components/mails_input.coffee]: No, you're right, it works in Firefox and Chromium but may fail in other engines, I'm gonna fix it.

**Pull request merged**